qiskit_metal/_gui/widgets/qlibrary_display/tree_view_qlibrary.py

```diff
@@ -0,0 +1,88 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Tree view for Param Entry Window
+"""
+
+from PySide2 import QtCore, QtWidgets, QtGui
+from PySide2.QtWidgets import QTreeView
+from PySide2.QtCore import QModelIndex, Signal
+from qiskit_metal._gui.widgets.qlibrary_display.proxy_model_qlibrary import LibraryFileProxyModel
+
+
+class TreeViewQLibrary(QTreeView):
+    """Handles editing a QComponent
+
+    This class extend the `QTreeView`
+    """
+
+    qlibrary_rebuild_signal = Signal(str)
+    qlibrary_filepath_signal = Signal(str)
+
+    def __init__(self, parent: QtWidgets.QWidget):
+        """
+        Args:
+            parent (QtWidgets.QWidget): parent widget
+        """
+        QTreeView.__init__(self, parent)
+        self.is_dev_mode = False  # Whether MetalGUI is in developer mode
+
+    def set_dev_mode(self, ison: bool):
+        """ Sets dev mode for self, model, model's source model, and delegate """
+        self.is_dev_mode = ison
```

Comment:
Throughout this file:

- Update docstring to include the return type
- Update args to include argument types
- Update docstring to include exceptions